source/backend/api/Areas/Documents/DocumentController.cs

```diff
@@ -142,6 +142,19 @@ public async Task<IActionResult> GetDocumentStorageTypeMetadata(long mayanDocume
             return new JsonResult(result);
         }
 
+        /// <summary>
+        /// Retrieves the document detail for a document type.
+        /// </summary>
+        [HttpGet("storage/{mayanDocumentId}/detail")]
+        [HasPermission(Permissions.DocumentAdd)]
```

Comment:
Also, could you please confirm that **DocumentAdd** is correct here? I would think **DocumentRead** would be instead...